### PR TITLE
Update Action Mailer Basics [ci skip]

### PR DESCRIPTION
**Pull request by aalvrz**
### Summary

Includes the [ActionMailer::Parameterized](http://edgeapi.rubyonrails.org/classes/ActionMailer/Parameterized.html) feature in the [Action Mailer Basics guide](http://guides.rubyonrails.org/action_mailer_basics.html) with an example.

### CODE REVIEW

File: guides/source/action_mailer_basics.md [hunk 1]
```diff
@@ -723,6 +723,38 @@ end
 
 * Mailer Filters abort further processing if body is set to a non-nil value.
 
+### Parameterized Mailers
+
+Instance variables, processing, and common headers can be shared by using [parameterized mailers](http://edgeapi.rubyonrails.org/classes/ActionMailer/Parameterized.html):
```
Comment (kaspth) [inline guides/source/action_mailer_basics.md]: This shouldn't link to `edgeapi` but just `api`.

**Review by kaspth — COMMENTED** [review]
I think @dhh meant parameterized mailers to be the recommended way to write mailers going forward (though not a full on default).

Can we integrate parameterized mailers into the guide more? Let's try a version that only uses `with` for instance.

**Comment by BigChief45** [comment]
> Can we integrate parameterized mailers into the guide more? Let's try a version that only uses with for instance.

@kaspth Do you mean we should change all the guide's example to use `with`? Not sure if I really understand.

**Comment by kaspth** [comment]
I mean the example mailers in the guide, like this one:

```ruby
class UserMailer < ApplicationMailer
  default from: 'notifications@example.com'

  def welcome_email(user)
    @user = user
    @url  = 'http://example.com/login'
    mail(to: @user.email, subject: 'Welcome to My Awesome Site')
  end
end
```

Should be changed to:

```ruby
class UserMailer < ApplicationMailer
  default from: 'notifications@example.com'

  def welcome_email
    @user = params[:user]
    @url  = 'http://example.com/login'
    mail(to: @user.email, subject: 'Welcome to My Awesome Site')
  end
end
```

We needn't use `before_action` for these simple cases. And then yes, use the `with` method to call the mailers.

We should also move the `InvitationsMailer` example to before the `after_action` example (the `after_action` looks like it could gain from using `params` + `before_action` as well).

How does that sound? :)

**Comment by kaspth** [comment]
Oh, and `default` calls with `Proc.new` should be changed to use `->`.

**Comment by BigChief45** [comment]
@kaspth 

I have done the changes you requested. I also deleted the section I added in the 1st commit for the *Action Mailer Parameterized* feature.

**Comment by BigChief45** [comment]
@rafaelfranca

File: guides/source/action_mailer_basics.md [hunk 2]
```diff
@@ -220,7 +220,7 @@ If you want to send emails right away (from a cronjob for example) just call
 class SendWeeklySummary
   def run
     User.find_each do |user|
-      UserMailer.weekly_summary(user).deliver_now
+      UserMailer.weekly_summary.with(user: user).deliver_now
```
Comment (kaspth) [inline guides/source/action_mailer_basics.md]: It's `with(user: user).weekly_summary` etc.

File: guides/source/action_mailer_basics.md [hunk 3]
```diff
@@ -677,24 +677,44 @@ Action Mailer allows for you to specify a `before_action`, `after_action` and
...
+
   after_action :set_delivery_options,
                :prevent_delivery_to_guests,
                :set_business_headers
 
-  def feedback_message(business, user)
-    @business = business
-    @user = user
+  def feedback_message
     mail
```
Comment (kaspth) [inline guides/source/action_mailer_basics.md]: No need for calling `mail` manually here. Rails handles that.

**Comment by kaspth** [comment]
Remember to keep your commits squashed to 1 as well.

**Comment by BigChief45** [comment]
@kaspth I have made the extra changes requested and squashed to 1 commit.

**Comment by kaspth** [comment]
Thanks!

**Pull request merged**